lib/x509.mli

```diff
@@ -69,12 +69,45 @@ module Host : sig
   end
 end
 
+module Dsa_curves : sig
+  module type Dsa = Mirage_crypto_ec.Dsa
+  module type S = Dsa_curves.S
+
+  type t = (module S)
+
+  module OIDs : sig
+    val secp224r1 : Asn.oid
+    val secp256r1 : Asn.oid
+    val secp384r1 : Asn.oid
+    val secp521r1 : Asn.oid
+    val secp256k1 : Asn.oid
+    val brainpoolP160r1 : Asn.oid
+    val brainpoolP160t1 : Asn.oid
+    val brainpoolP192r1 : Asn.oid
+    val brainpoolP192t1 : Asn.oid
+    val brainpoolP224r1 : Asn.oid
+    val brainpoolP224t1 : Asn.oid
+    val brainpoolP256r1 : Asn.oid
+    val brainpoolP256t1 : Asn.oid
+    val brainpoolP320r1 : Asn.oid
+    val brainpoolP320t1 : Asn.oid
+    val brainpoolP384r1 : Asn.oid
+    val brainpoolP384t1 : Asn.oid
+    val brainpoolP512r1 : Asn.oid
+    val brainpoolP512t1 : Asn.oid
+  end
+
+  val register :
+    string -> Asn.oid -> (module Dsa) -> t
```

Comment:
why return a t from register? shouldn't it just be a unit? or what is it that you like to do with t? (also, taking the other comment into account, should it possible to return an error to avoid multiple registrations of the same curve?)